CAP/gap/TerminalCategory.gi

```diff
@@ -4,148 +4,416 @@
 # Implementations
 #
 
-#####################################
-##
-## Reps for object and morphism
-##
-#####################################
-
-DeclareRepresentation( "IsCapTerminalCategoryObjectRep",
-                       IsAttributeStoringRep and IsCapCategoryObjectRep,
-                       [ ] );
-
-DeclareRepresentation( "IsCapTerminalCategoryMorphismRep",
-                       IsAttributeStoringRep and IsCapCategoryMorphismRep,
-                       [ ] );
-
-BindGlobal( "TheTypeOfCapTerminalCategoryObject",
-        NewType( TheFamilyOfCapCategoryObjects,
-                IsCapTerminalCategoryObjectRep ) );
-
-BindGlobal( "TheTypeOfCapTerminalCategoryMorphism",
-        NewType( TheFamilyOfCapCategoryMorphisms,
-                IsCapTerminalCategoryMorphismRep ) );
-
-#####################################
-##
-## Constructor
-##
-#####################################
-
-InstallValue( CAP_INTERNAL_TERMINAL_CATEGORY,
-              
-              CreateCapCategory( "TerminalCategory" ) );
-
-SetFilterObj( CAP_INTERNAL_TERMINAL_CATEGORY, IsTerminalCategory );
-
-InstallValue( CAP_INTERNAL_TERMINAL_CATEGORY_AS_CAT_OBJECT,
-              
-              AsCatObject( CAP_INTERNAL_TERMINAL_CATEGORY ) );
-
-##
-InstallMethod( UniqueObject,
-               [ IsCapCategory and IsTerminalCategory ],
-               
-  function( category )
-    local object;
+InstallGlobalFunction( CAP_INTERNAL_CONSTRUCTOR_FOR_TERMINAL_CATEGORY,
+  function( input_record )
+    local completed_record, category_filter, create_func_bool,
+          list_of_operations_to_install, operation_name, info, with_given_object_name,
+          r, func, pos, properties, excluded_properties, T;
+    
+    completed_record := ShallowCopy( input_record );
+    
+    ## e.g., IsMonomorphism
+    create_func_bool :=
+        function( operation_name, T )
+            
+            return """
+                function( input_arguments... )
+                    
+                    return true;
+                    
+                end
+            """;
+            
+        end;
+    
+    completed_record.create_func_bool := create_func_bool;
+    
+    ## FIXME: operations should be installed based on doctrines (as far as possible)
+    list_of_operations_to_install :=
+      Concatenation( List( RecNames( CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD ), p -> CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD.(p) ) );
+    
+    for r in
+      [ "RIGID_SYMMETRIC_CLOSED_MONOIDAL_CATEGORIES_METHOD_NAME_RECORD",
+        "RIGID_SYMMETRIC_COCLOSED_MONOIDAL_CATEGORIES_METHOD_NAME_RECORD",
+        "CATEGORY_CONSTRUCTOR_METHOD_NAME_RECORD",
+        "FREYD_CATEGORIES_METHOD_NAME_RECORD",
+        "BOOLEAN_ALGEBRA_METHOD_NAME_RECORD",
+        ] do
+        
+        if IsBoundGlobal( r ) then
+            Append( list_of_operations_to_install, RecNames( ValueGlobal( r ) ) );
+        fi;
+        
+    od;
```

Comment:
I'm not sure what the result of our last discussion was: Do we want to also remove these hardcoded values?

Reply:
I think we only briefly talked about the skipped operations above, but not about these. :/

Reply:
Then let's remove this list (making sure the tests relying in TerminalCategory still pass). Then we can reintroduce what is needed in the future using an improved system.